homeassistant/components/thingspeak.py

```diff
@@ -43,7 +43,7 @@ def setup(hass, config):
 
     try:
         channel = thingspeak.Channel(
-            channel_id, api_key=api_key, timeout=TIMEOUT)
+            channel_id, write_key=api_key, timeout=TIMEOUT)
```

Comment:
No, sorry, you are wrong. Check your [code](https://github.com/mchwalisz/thingspeak/blob/master/thingspeak/thingspeak.py#L103). `api_key`needs to be part of the payload for the [request](https://ch.mathworks.com/help/thingspeak/update-channel-feed.html). You changed it in `develop` but 0.4.1 is still using `api_key` and `write_key`. 

Want prove?

```bash
$ pip3 freeze | grep thingspeak
thingspeak==0.4.1
(home-assistant) [fab@laptop016 home-assistant]$ python3
Python 3.5.2 (default, Oct 14 2016, 12:54:53) 
[GCC 6.2.1 20160916 (Red Hat 6.2.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import thingspeak
>>> import random
>>> api_key = 'WRITE_API_KEY_FOR_THE_CHANNEL'
>>> channel_id = 219860
>>> channel = thingspeak.Channel(channel_id, api_key=api_key, timeout=5)
>>> channel.update({'field1': random.randrange(2, 29)})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/fab/Documents/repos/home-assistant/lib64/python3.5/site-packages/thingspeak/thingspeak.py", line 117, in update
    return self._fmt(r)
  File "/home/fab/Documents/repos/home-assistant/lib64/python3.5/site-packages/thingspeak/thingspeak.py", line 120, in _fmt
    r.raise_for_status()
  File "/home/fab/Documents/repos/home-assistant/lib64/python3.5/site-packages/requests/models.py", line 909, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.thingspeak.com//update.json?field1=27
>>> channel = thingspeak.Channel(channel_id, write_key=api_key, timeout=5)
>>> channel.update({'field1': random.randrange(2, 29)})
'{"channel_id":219860,"field1":"14","field2":null,"field3":null,"field4":null,"field5":null,"field6":null,"field7":null,"field8":null,"created_at":"2017-02-12T14:02:03Z","entry_id":143,"status":null,"latitude":null,"longitude":null,"elevation":null}'
```